src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -304,16 +304,21 @@ public void WriteToStream(Stream stream)
         public async void WriteToStreamAsync(Stream stream, SNIAsyncCallback callback, SNIProviders provider)
         {
             uint status = TdsEnums.SNI_SUCCESS;
-            try
-            {
-                await stream.WriteAsync(_data, 0, _dataLength, CancellationToken.None).ConfigureAwait(false);
-            }
-            catch (Exception e)
+
+            await stream.WriteAsync(_data, 0, _dataLength, CancellationToken.None).ContinueWith(t =>
             {
-                SNILoadHandle.SingletonInstance.LastError = new SNIError(provider, SNICommon.InternalExceptionError, e);
-                status = TdsEnums.SNI_ERROR;
-            }
-            callback(this, status);
+                Exception e = t.Exception?.InnerException;
+                if (e != null)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(provider, SNICommon.InternalExceptionError, e);
+                    status = TdsEnums.SNI_ERROR;
+                    Release();
```

Comment:
Just wanted to callout "Release()" added here, I guess this is made intentionally?

Reply:
I couldn't see a reason except resue the packet's allocated memory after encountering by an exception.